### PR TITLE
Add Ingress example for Kubernetes 1.19+ and apiVersion: "networking.k8s.io/v1"

### PR DESCRIPTION
**Pull request by sturman**
# What this PR does / why we need it
This PR adds example how to configure paths in Ingress object with `apiVersion: "networking.k8s.io/v1"` that used in Kubernetes 1.19+
Ingress 

# Checklist

- [x] [DCO](https://github.com/jenkinsci/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] CHANGELOG.md was updated

### CODE REVIEW

File: charts/jenkins/values.yaml [hunk 1]
```diff
@@ -343,8 +343,14 @@ controller:
 
   ingress:
     enabled: false
+    # For Kubernetes v1.14+, use 'networking.k8s.io/v1beta1'
```
Comment (timja) [inline charts/jenkins/values.yaml:346]: to be honest this isn't really true we should probably remove this flag?

the version is hardcoded except for <= 1.13.0 which is long gone
https://github.com/jenkinsci/helm-charts/blob/main/charts/jenkins/templates/jenkins-controller-ingress.yaml#L3-L9

Comment (sturman) [reply charts/jenkins/values.yaml]: Updated to 
- `For Kubernetes >=v1.14 and <v1.19, use 'networking.k8s.io/v1beta1'`
- `For Kubernetes v1.19+, use 'networking.k8s.io/v1'`

Comment (timja) [reply charts/jenkins/values.yaml]: I mean that comment is pointless, the code doesn't actually use what you point there, the proper fix would be to remove the version from there, and hardcode the old version in the jenkins-controller-ingress.yaml

Comment (sturman) [reply charts/jenkins/values.yaml]: Looks like the code does use the apiVersion specified in values https://github.com/jenkinsci/helm-charts/blob/main/charts/jenkins/templates/jenkins-controller-ingress.yaml#L8
Or am I missing something?

Comment (timja) [reply charts/jenkins/values.yaml]: it only uses it on <= 1.13.0, check the lines above it. it should be inlined there

File: charts/jenkins/templates/jenkins-controller-ingress.yaml [hunk 2]
```diff
@@ -1,12 +1,6 @@
 {{- $kubeTargetVersion := default .Capabilities.KubeVersion.GitVersion .Values.kubeTargetVersionOverride }}
 {{- if .Values.controller.ingress.enabled }}
-{{- if semverCompare ">=1.19-0" $kubeTargetVersion -}}
-apiVersion: networking.k8s.io/v1
```
Comment (timja) [inline charts/jenkins/templates/jenkins-controller-ingress.yaml:4]: this isn't what i meant, i meant just remove it from values.yaml, there's no reason to have it configurable

**Pull request closed without merging**